opte/src/engine/int_test.rs

```diff
@@ -163,18 +395,41 @@ fn oxide_net_builder(name: &str, cfg: &VpcCfg) -> PortBuilder {
     pb
 }
 
-fn oxide_net_setup(name: &str, cfg: &VpcCfg) -> Port {
-    oxide_net_builder(name, cfg).create(UFT_LIMIT.unwrap(), TCP_LIMIT.unwrap())
+struct PortAndVps {
+    port: Port,
+    vps: VpcPortState,
+}
+
+fn oxide_net_setup(name: &str, cfg: &VpcCfg) -> PortAndVps {
+    let port = oxide_net_builder(name, cfg)
+        .create(UFT_LIMIT.unwrap(), TCP_LIMIT.unwrap());
+    assert_eq!(port.state(), PortState::Ready);
+    assert_eq!(port.epoch(), 1);
+    check_no_flows(&port);
+    let vps = VpcPortState::new();
+    let mut pav = PortAndVps { port, vps };
+    update!(
+        pav,
+        [
+            "set:arp.rules_in=1,arp.rules_out=2",
```

Comment:
So I admit I'm a bit confused about these fields. It seems like we're counting the rules or flows for the port, is that right? Making this separate seems like we have two sources of truth, though. I'd have thought the counts could be either updated automatically as rules or flows are added or removed; or they'd be computed on the fly whenever we need them.

I guess I mean, why do we want the counts in the `VpcPortState` field, distinct from whatever's added to the `Port` itself?

Reply:
Ok, I _completely_ missed the fact that this is a test-only file. I'm betting that answers my question, and sorry about that!

Reply:
Yea, sorry, I didn't really write up any comments in the file about what this is call for. The point of VpcPortState and its macros is to drastically cut down the redundancy in the tests around the checking of various port state like number of rules, flows, and the current epoch. I'll add a comment.